python/ray/scripts/scripts.py

```diff
@@ -519,16 +521,25 @@ def start(node_ip_address, address, port, redis_password, redis_shard_ports,
         no_monitor=no_monitor,
         tracing_startup_hook=tracing_startup_hook,
         ray_debugger_external=ray_debugger_external)
+
     if head:
-        # Use default if port is none, allocate an available port if port is 0
+        # Start head node.
+
         if port is None:
             port = ray_constants.DEFAULT_PORT
-
+        # TODO(mwtian): use a more robust mechanism to avoid collision,
+        # e.g. node._get_cached_port()
         if port == 0:
             with socket() as s:
                 s.bind(("", 0))
                 port = s.getsockname()[1]
 
+        # Set bootstrap port.
+        ray_params.redis_port = port
```

Comment:
do we need to skip this set if it's using gcs bootstrap? if we'll fix this in next PR, please add TODO